docs/running-on-yarn.md

```diff
@@ -134,6 +134,13 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   </td>
 </tr>
 <tr>
+  <td><code>spark.yarn.am.port</code></td>
+  <td>(random)</td>
+  <td>
+    Port for the YARN Application Master to listen on. This is used for communicating with the executors.
```

Comment:
this is not the channel used to communicate with executors.
This channel is used in yarn client mode to communicate between the spark driver running on a gateway and the application master (which is doing container requests) running on yarn.

It also looks like I was wrong  my previous comment. It actually is also used in yarn cluster mode.  Its used for the dynamic executor feature, where it is handling the kill from the scheduler backend.